homeassistant/components/switcher_kis/_service_registration.py

```diff
@@ -0,0 +1,371 @@
+})
+
+SERVICE_CREATE_SCHEDULE = 'create_schedule'
+CREATE_RECURRING_SCHEDULE_SERVICE_SCHEMA = vol.All(vol.Schema({
+    vol.Required(CONF_START_TIME): cv.time_period_str,
+    vol.Required(CONF_END_TIME): cv.time_period_str,
+    vol.Required(CONF_RECURRING): vol.All(cv.boolean, True),
+    vol.Optional(CONF_DAYS, default=DEFAULT_CONF_DAYS): vol.All(
+        cv.ensure_list_csv, [vol.In([MONDAY, TUESDAY, WEDNESDAY,
+                                     THURSDAY, FRIDAY, SATURDAY, SUNDAY])]
+        )
+}))
+
+CREATE_NON_RECURRING_SCHEDULE_SERVICE_SCHEMA = vol.All(vol.Schema({
+    vol.Required(CONF_START_TIME): cv.time_period_str,
+    vol.Required(CONF_END_TIME): cv.time_period_str,
+    vol.Required(CONF_RECURRING): vol.All(cv.boolean, False),
+    vol.Optional(CONF_DAYS, default=DEFAULT_CONF_DAYS): vol.Any(
+        cv.match_all, None)
+}))
+
+CREATE_SCHEDULE_SERVICE_SCHEMA = vol.Any(
+    CREATE_RECURRING_SCHEDULE_SERVICE_SCHEMA,
+    CREATE_NON_RECURRING_SCHEDULE_SERVICE_SCHEMA)
+
+
+SWITCH_ENTITY = None  # type: Any
+SENSOR_ENTITIES = {}  # type: Dict[str, SwitcherV2Schedule]
+
+
+async def async_register_switch_entity(switch_created: SwitchDevice) -> None:
+    """Use for registering the switch entity for data updates."""
+    global SWITCH_ENTITY
+    SWITCH_ENTITY = switch_created
+
+
+async def async_register_sensor_entities(sensors_created: Dict[str, Entity]) \
+        -> None:
+    """Use for registering the sensor entity for data updates."""
+    global SENSOR_ENTITIES
+    SENSOR_ENTITIES = sensors_created
+
+
+async def async_reg_services_platforms(
+        hass: HomeAssistant, config: Dict, phone_id: str, device_id: str,
+        device_password: str, include_sensors: bool,
+        schedules_scan_interval: timedelta) -> None:
+    """Use for registering HA's services and listen for platforms."""
+    async def async_update_device_cb(device_data: SwitcherV2Device) -> None:
+        """Use this as callback for updating the switcher entity.
+
+        Called by the bridge thread on recursive broadcast message.
+        """
+        if SWITCH_ENTITY:
+            await SWITCH_ENTITY.async_update_data(device_data)
+
+        return None
+
+    async def async_initial_data_cb(device_data: SwitcherV2Device) -> None:
+        """Use this as callback for creating home assistant's entities.
+
+        Called by the bridge thread on initial device discovery.
+        """
+        discovery_info = {
+            DISCOVERY_CONFIG: config[DOMAIN],
+            DISCOVERY_DEVICE: device_data
+        }
+
+        # CONF_SCHEDULE_SCAN_INTERVAL is a timedelta and not serializable.
+        discovery_info[DISCOVERY_CONFIG].pop(CONF_SCHEDULE_SCAN_INTERVAL, None)
+
+        await hass.async_create_task(
+            async_load_platform(
+                hass, SWITCH_DOMAIN, DOMAIN, discovery_info, config))
+
+        if include_sensors:
+            response = await get_schedules(device_data.ip_addr, phone_id,
```

Comment:
While defined as `async` the implementation of this method calls blocking synchronous methods. This will need to be fixed in the library.